src/org/omegat/core/team2/impl/GITCredentialsProvider.java

```diff
@@ -247,54 +228,89 @@ public boolean get(URIish uri, CredentialItem... items) throws UnsupportedCreden
                 // RSA key fingerprint is e2:d3:84:d5:86:e7:68:69:a0:aa:a6:ad:a3:a0:ab:a2.
                 // Are you sure you want to continue connecting?
                 String storedFingerprint = loadFingerprint(uri);
-                String promptText = i.getPromptText();
+                String promptText = sb.append(i.getPromptText()).toString();
+                sb = new StringBuilder();
                 String promptedFingerprint = extractFingerprint(promptText);
                 if (promptedFingerprint == null) {
                     throw new UnsupportedCredentialItem(uri, "Wrong fingerprint pattern");
                 }
                 if (predefinedFingerprint != null) {
-                    if (promptedFingerprint.equals(predefinedFingerprint)) {
-                        ((CredentialItem.YesNoType) i).setValue(true);
-                    } else {
-                        ((CredentialItem.YesNoType) i).setValue(false);
-                    }
+                    ((CredentialItem.YesNoType) i).setValue(promptedFingerprint.equals(predefinedFingerprint));
                     continue;
                 }
                 if (promptedFingerprint.equals(storedFingerprint)) {
                     ((CredentialItem.YesNoType) i).setValue(true);
                     continue;
                 }
-                int choice = Core.getMainWindow().showConfirmDialog(promptText, null,
-                        JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
-                if (choice == JOptionPane.YES_OPTION) {
-                    ((CredentialItem.YesNoType) i).setValue(true);
-                    saveFingerprint(uri, promptedFingerprint);
+                if (mw != null) {  // on GUI application
+                    int choice = mw.showConfirmDialog(promptText, null,
+                            JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
+                    if (choice == JOptionPane.YES_OPTION) {
+                        ((CredentialItem.YesNoType) i).setValue(true);
+                        saveFingerprint(uri, promptedFingerprint);
+                    } else {
+                        ((CredentialItem.YesNoType) i).setValue(false);
+                    }
                 } else {
-                    ((CredentialItem.YesNoType) i).setValue(false);
+                    Console console = System.console();
+                    if (console != null) {
+                        PrintWriter printWriter = console.writer();
+                        int j = 0;
+                        while (j++ < 5) {
```

Comment:
I feel like this would be a bit nicer as a `for` loop. Instead of looking at the value of `j` to determine whether to abort, use a boolean flag like `boolean succeeded = false;` and set it to true on either of the success branches.